### PR TITLE
LPD-31974 - ERC Support for specifications (specification labels) follow up

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-31974

### CODE REVIEW

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java [hunk 1]
```diff
@@ -125,8 +125,11 @@ public Specification patchSpecification(
 			Long id, Specification specification)
 		throws Exception {
 
-		CPSpecificationOption cpSpecificationOption = _updateSpecification(
-			id, specification);
+		CPSpecificationOption cpSpecificationOption =
+			_cpSpecificationOptionService.getCPSpecificationOption(id);
+
+		cpSpecificationOption = _updateSpecification(
+			specification, cpSpecificationOption);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java:132]: what about
```
CPSpecificationOption cpSpecificationOption =
			_updateSpecification(cpSpecificationOption, _cpSpecificationOptionService.getCPSpecificationOption(id));
```
?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java [hunk 2]
```diff
@@ -165,7 +168,11 @@ private Specification _addOrUpdateSpecification(Specification specification)
 		if (specificationId != null) {
 			try {
 				CPSpecificationOption cpSpecificationOption =
-					_updateSpecification(specificationId, specification);
+					_cpSpecificationOptionService.getCPSpecificationOption(
+						specificationId);
+
+				cpSpecificationOption = _updateSpecification(
+					specification, cpSpecificationOption);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java:175]: same, inline them

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java [hunk 3]
```diff
@@ -182,12 +189,49 @@ private Specification _addOrUpdateSpecification(Specification specification)
...
-		if (specificationKey != null) {
+		if (Validator.isNotNull(specificationKey)) {
 			try {
 				CPSpecificationOption cpSpecificationOption =
-					_updateSpecification(specificationKey, specification);
+					_cpSpecificationOptionService.getCPSpecificationOption(
+						contextCompany.getCompanyId(), specificationKey);
+
+				cpSpecificationOption = _updateSpecification(
+					specification, cpSpecificationOption);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java:234]: inline them

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java [hunk 4]
```diff
@@ -285,26 +329,20 @@ private CPSpecificationOption _updateSpecification(
 	}
 
 	private CPSpecificationOption _updateSpecification(
-			String key, Specification specification)
+			Specification specification,
+			CPSpecificationOption cpSpecificationOption)
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SpecificationResourceImpl.java:333]: please invert the parameter (to be alphabetical ordered)

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**